barretenberg/cpp/src/barretenberg/client_ivc/client_ivc.cpp

```diff
@@ -245,6 +245,19 @@ std::pair<std::shared_ptr<ClientIVC::DeciderZKProvingKey>, ClientIVC::MergeProof
     fold_output.accumulator = nullptr;
 
     ClientCircuit builder{ goblin.op_queue };
+
+    // Translator circuit builder computes the evaluation at some random challenge x of a batched polynomial derived
```

Comment:
Can you make this a method and add this detailed description to that method? This construct_hiding_circuit_key method already contains more complicated logic than one method should.

Reply:
good point!